### PR TITLE
feat: UI knows and reacts according to the event state

### PR DESCRIPTION
**Pull request by jorgenavben**

### CODE REVIEW

File: ui/summit-2024/src/components/ConnectWalletModal/ConnectWalletModal.tsx [hunk 1]
```diff
@@ -244,7 +244,7 @@ const ConnectWalletModal = (props: ConnectWalletProps) => {
         };
       case ConnectWalletFlow.CONNECT_IDENTITY_WALLET:
         return {
-          title: "Use IDW Wallet to connect",
+          title: "Use ID Wallet to connect",
```
Comment (iFergal) [inline ui/summit-2024/src/components/ConnectWalletModal/ConnectWalletModal.tsx:247]: these also should all be in i18n - maybe we can just create a catch all ticket to convert these last minute if we have time.

File: ui/summit-2024/src/pages/Leaderboard/Leaderboard.tsx [hunk 2]
```diff
@@ -106,12 +113,32 @@ const Leaderboard: React.FC = () => {
...
                 Leaderboard
               </Typography>
-              <AnimatedSwitch
-                defaultValue="Overall Votes"
-                optionA="Winners"
-                optionB="Overall Votes"
-                onClickOption={handleSwitch}
-              />
+              
+              {showRevealDate && ( // Shows once the voting has finished and before the reveal
```
Comment (iFergal) [inline ui/summit-2024/src/pages/Leaderboard/Leaderboard.tsx:117]: So both `showRevealDate` and `showWinners` are boolean values. When you use them with && in JSX when rendering, if they are false I think it will display a `0` randomly on the page. Safer (but not as nice) to use `showRevealDate ? (...) : undefined` (or null)

File: ui/summit-2024/src/pages/Leaderboard/Leaderboard.tsx [hunk 3]
```diff
@@ -106,12 +113,32 @@ const Leaderboard: React.FC = () => {
...
+                <Typography // TODO: Format
```
Comment (iFergal) [inline ui/summit-2024/src/pages/Leaderboard/Leaderboard.tsx:118]: todo?

File: ui/summit-2024/src/pages/Categories/components/NomineeCard.tsx [hunk 4]
```diff
@@ -61,7 +66,7 @@ const NomineeCard: React.FC<NomineeCardProps> = ({
           flexShrink: 0,
           borderRadius: "24px",
           border: `1px solid ${
-            selectedNominee === nominee.id || votedNominee
+            (selectedNominee === nominee.id || votedNominee) && showSelection
```
Comment (iFergal) [inline ui/summit-2024/src/pages/Categories/components/NomineeCard.tsx:69]: If they aren't allowed to vote, `allowToVote=false` as their wallet isn't verified, should be they be able to click cards?

File: ui/summit-2024/src/pages/Categories/components/NomineeCard.tsx [hunk 5]
```diff
@@ -82,13 +87,16 @@ const NomineeCard: React.FC<NomineeCardProps> = ({
...
           }}
         >
-          <Box component="div" sx={{ position: "absolute", right: 8, top: 8 }}>
-            {allowToVote || votedNominee ? (
-              <HoverCircle
-                selected={selectedNominee === nominee.id || votedNominee}
-              />
-            ) : null}
-          </Box>
+          {showSelection && (
```
Comment (iFergal) [inline ui/summit-2024/src/pages/Categories/components/NomineeCard.tsx:90]: same here re && and ? : operators

File: ui/summit-2024/src/pages/Home/components/Hero.tsx [hunk 6]
```diff
@@ -62,7 +64,14 @@ const Hero = () => {
...
           >
-            Voting Closes {formatISODate(eventCache.eventEndDate)}
+            {eventCache.notStarted
+              ? "Voting Opens " + formatISODate(eventCache.eventStartDate)
+              : eventCache.active
+                ? "Voting Closes " + formatISODate(eventCache.eventEndDate)
+                : !eventCache.proposalsReveal
+                  ? "Results Announced " +
+                    formatISODate(eventCache.proposalsRevealDate)
+                  : "THE RESULTS ARE IN!!!"}
```
Comment (iFergal) [inline ui/summit-2024/src/pages/Home/components/Hero.tsx:74]: Might be clearer to have a little function for this:
```
const getVotingText = () => {
  if (eventCache.notStarted) {
    return "Voting Opens " + formatISODate(eventCache.eventStartDate);
  }
  if (eventCache.active) {
    return "Voting Closes " + formatISODate(eventCache.eventEndDate);
  }
  if (eventCache.proposalsReveal) {
    return "THE RESULTS ARE IN!!!"
  }
  return "Results Announced " + formatISODate(eventCache.proposalsRevealDate);
};
```

Sometimes can be hard to follow chained ternary operators, especially when one is negated (`!eventCache.proposalsReveal`)

Comment (iFergal) [inline ui/summit-2024/src/pages/Home/components/Hero.tsx:74]: Is it actually meant to be "THE RESULTS ARE IN!!!"?

File: ui/summit-2024/src/pages/Categories/Categories.tsx [hunk 7]
```diff
@@ -359,18 +364,35 @@ const Categories: React.FC<CategoriesProps> = ({ embedded }) => {
...
-                display: isTablet ? "none" : "inline-block",
-              }}
-              colorVariant="primary"
-              disabled={renderActionButton().disabled}
-            >
-              {renderActionButton().label}
-            </CustomButton>
+
+            {showEventDate && ( // If the event has not started or it's just before the reveal
+              <Typography  // TODO: Formatting
```
Comment (iFergal) [inline ui/summit-2024/src/pages/Categories/Categories.tsx:369]: todo

File: ui/summit-2024/src/pages/Categories/Categories.tsx [hunk 8]
```diff
@@ -359,18 +364,35 @@ const Categories: React.FC<CategoriesProps> = ({ embedded }) => {
...
+                }}
+              >
+                {eventCache.notStarted
+                  ? "Voting Opens " + formatISODate(eventCache.eventStartDate)
+                  : "Results Announced " +
+                    formatISODate(eventCache.proposalsRevealDate)}
+              </Typography>
+            )}
+
+            {showVotingButton && ( // If the event has not started or the results have been revealed we prevent the voting bottom to show up
```
Comment (iFergal) [inline ui/summit-2024/src/pages/Categories/Categories.tsx:382]: && vs ? : again

File: ui/summit-2024/src/pages/Categories/Categories.tsx [hunk 9]
```diff
@@ -359,18 +364,35 @@ const Categories: React.FC<CategoriesProps> = ({ embedded }) => {
...
-                alignSelf: "flex-end",
-                display: isTablet ? "none" : "inline-block",
-              }}
-              colorVariant="primary"
-              disabled={renderActionButton().disabled}
-            >
-              {renderActionButton().label}
-            </CustomButton>
+
+            {showEventDate && ( // If the event has not started or it's just before the reveal
```
Comment (iFergal) [inline ui/summit-2024/src/pages/Categories/Categories.tsx:368]: && vs ? :

File: ui/summit-2024/src/pages/Categories/components/NomineeCard.tsx (hunk 4 above)
Comment (jorgenavben) [reply ui/summit-2024/src/pages/Categories/components/NomineeCard.tsx:69]: According to [this design](https://www.figma.com/design/cRGPl1mAbAsY9MUuJM14xf/Cardano-Ballot-2024?node-id=452-6144&t=5I72gL5oWCUqRzPp-4) from Robert they can click on the the cards

File: ui/summit-2024/src/pages/Home/components/Hero.tsx (hunk 6 above)
Comment (jorgenavben) [reply ui/summit-2024/src/pages/Home/components/Hero.tsx:74]: It also sounds strange to me but I got it from the [Figma designs](https://www.figma.com/design/cRGPl1mAbAsY9MUuJM14xf/Cardano-Ballot-2024?node-id=1222-5339&t=5I72gL5oWCUqRzPp-4)

File: ui/summit-2024/src/pages/Leaderboard/Leaderboard.tsx (hunk 2 above)
Comment (iFergal) [reply ui/summit-2024/src/pages/Leaderboard/Leaderboard.tsx]: `showWinners` part below unchanged

**Review by iFergal — APPROVED** [review]
LGTM! you need to merge feature/ui into your branch first tho

File: ui/summit-2024/src/pages/Leaderboard/Leaderboard.tsx (hunk 2 above)
Comment (jorgenavben) [reply ui/summit-2024/src/pages/Leaderboard/Leaderboard.tsx]: My bad. Now it has been fixed
